pkg/scheduler/api/types.go

```diff
@@ -23,6 +23,12 @@ const (
 	// Pending means the task is pending in the apiserver.
 	Pending TaskStatus = 1 << iota
 
+	// AllocatedOverBackfill means that the task is allocated with resources
+	// that are currently occupied by backfill tasks.
+	// Task T is allocated to Node N as an AllocatedOverBackfill task if,
+	// and only if, N.IdleResource < T.RequestResource <= N.AllocatableResource.
+	AllocatedOverBackfill
```

Comment:
Can we reuse `Pipelined`? If not, prefer `Waiting` which means waiting for running task to release resources.

Reply:
AllocatedOverBackfill cannot be replaced with Pipelined. Tasks allocated with AllocatedOverBackfill status will release their resources in allocation if their belonging jobs are still not ready. We do not do the same for pipelined task.

Reply:
On the other hand, "Waiting" does not have a clear meaning on what's being waited for. What about "AllocatedWithBackfillResource". It's long but I'd trade length for confusion.

Reply:
I do not want to add a phase for just one feature.

Reply:
renamed to AllocatedPendingResource

Reply:
> renamed to AllocatedPendingResource

Try to use one word, similar to other status.

Reply:
One word will cause confusion and why do we want to enforce that?

Reply:
the closest I can think of is "Expecting"

Reply:
renamed to "Borrowing"